src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -723,4 +767,365 @@ export class EditDataComponent extends GridParentComponent implements OnInit, On
 		}
 
 	}
+
+	//my code here:
+	private createNewTable(): void {
+
+		let t: Table<any>;
+		if (this.placeHolderDataSets) {
+			let dataSet = this.placeHolderDataSets[0];
+			let options = {
+				enableCellNavigation: true,
+				enableColumnReorder: false,
+				renderRowWithRange: true,
+				showHeader: true,
+				rowHeight: this._rowHeight,
+				defaultColumnWidth: 120,
+				editable: false,
+				autoEdit: false,
+				enableAddRow: false, // TODO change when we support enableAddRow
+				enableAsyncPostRender: false,
+				editorFactory: {
+					getEditor: (column: ISlickColumn<any>) => this.getColumnEditor(column)
+				},
+				formatterFactory: {
+					getFormatter: this.getFormatter
+				}
+			};
+
+			if (dataSet.columnDefinitions) {
+				t = new Table(this.nativeElement, { dataProvider: new AsyncDataProvider(dataSet.dataRows), columns: dataSet.columnDefinitions }, options);
+
+				this._tables[0] = t;
+				if (this.selectionModel) {
+					if (typeof this.selectionModel === 'object') {
+						t.grid.setSelectionModel(this.selectionModel);
+					} else if (typeof this.selectionModel === 'string' && Slick[this.selectionModel] && typeof Slick[this.selectionModel] === 'function') {
+						t.grid.setSelectionModel(Slick[this.selectionModel]);
+					} else {
+						console.error(`Tried to register selection model ${this.selectionModel},
+						but none was found to be attached to Slick Grid or it was not a function.
+						Please extend the Slick namespace with the selection model as a function before registering`);
+					}
+				}
+				for (let plugin of this.plugins[0]) {
```

Comment:
seems strange to iterate of `this.plugins[0]` instead of `this.plugins`?  I guess `plugins` is an array or arrays?  In which case, why just loop over the first elements items?

Reply:
In the current state, its an array containing a single array INSIDE it for the first element, I think it was used for handling multiple tables instead of the single table approach that we are supposed to use now